vignettes/custom_data_sources.Rmd

```diff
@@ -0,0 +1,77 @@
+e.g. this: `https://www.sciencebase.gov/catalogMaps/mapping/ows/54296bf0e4b0ad29004c2fbb`  
+not this: `https://www.sciencebase.gov/catalogMaps/mapping/ows/54296bf0e4b0ad29004c2fbb?service=wfs&request=getcapabilities&version=1.0.0`
+
+Since this is an advanced feature, it may be interesting to see what `geoknife` is doing, swith `verbose=FALSE` to `verbose=TRUE` in the following code to see the web service request being made when you execute this vignette.
+
+```{r stencil, echo=T, eval=T}
+gconfig(verbose=FALSE)
+stencil <- webgeom(url="https://www.sciencebase.gov/catalogMaps/mapping/ows/54296bf0e4b0ad29004c2fbb")
+
+stencil_geoms <- query(stencil, 'geoms')
+print(stencil_geoms)
+```
+
+Now we can select a WFS geometry layer (`geom`) and query for its attributes. 
+```{r attribute, echo=T, eval=T}
+geom(stencil) <- stencil_geoms[2]
+
+stencil_attributes <- query(stencil, 'attributes')
+print(stencil_attributes)
+```
+
+Now we can select an attribute and query for the values of that attribute. For this demo, we won't set the values, but rather just move forward using all polygons in this geom
+```{r value, echo=T, eval=T}
+attribute(stencil) <- stencil_attributes[2]
+
+print(query(stencil, 'values'))
+```
+
+Now we'll set up our webdata with an OPeNDAP service as it's url. Note that this can be any OPeNDAP service compatible with `geoknife` and the Geo Data Portal. What datasets are compatible is beyond the scope of this vignette, but is documented [here](https://my.usgs.gov/confluence/display/GeoDataPortal/Custom+Dataset+Use+Guidlines). The OPeNDAP dataset used here is from the [main USGS THREDDS archive](https://cida.usgs.gov/thredds) but many other OPeNDAP datasets can be used in the same way.
+
+```{r fabric, echo=T, eval=T}
+fabric <- webdata(url = 'https://cida.usgs.gov/thredds/dodsC/prism_v2', 
+                  variables = c('tmx', 'tmn', 'ppt'),
+                  times = as.POSIXct(c('2000-01-01', '2010-01-01')))
+print(fabric)
+```
+
+Note that if you don't know the valid variables and times for an OPeNDAP URL, you can use `query` to find them like this.
+```{r query_fabric, echo=T, eval=T, warning=F}
+fabric_variables <- query(fabric, "variables")
+fabric_times <- query(fabric, "times")
+
+print(paste(fabric@url, "has", 
+            paste(fabric_variables, collapse = ", "), 
+            "variables for the time range", fabric_times[1], 
+            "to", fabric_times[2]))
+```
+
+Now we can execute the `geoknife` job for the specified fabric and stencil. Note that we set `wait = TRUE` here so we can just wait for the result of the process. For this vignette, we'll just print out the names and size of the returned data.
+
+```{r execute, echo=T, eval=T}
+prism_yahara_result <- result(geoknife(stencil, fabric, wait = TRUE))
```

Comment:
I think this is ok, but is it a pattern we want to suggest people bake into their code? I wonder if you want to make a special note that you are kind of cooking up your own `str()` method since one exists. Would you prefer `str()` or `summary()` for this kind of thing?